engine/runtime/src/main/java/org/enso/interpreter/runtime/scope/ModuleScope.java

```diff
@@ -262,9 +271,15 @@ public Optional<Type> getType(String name) {
     return Optional.ofNullable(types.get(name));
   }
 
-  /** @return the raw method map held by this module */
-  public Map<Type, Map<String, Function>> getMethods() {
-    return methods;
+  /** @return a map of methods for the given type */
+  public Map<String, Function> getMethodsForType(Type tpe) {
+    Type tpeKey = tpe == null ? noTypeKey : tpe;
+    return methods.get(tpe);
+  }
+
+  /** @return methods mappings for all registered types */
+  public Collection<Map<String, Function>> getAllMethods() {
```

Comment:
This is a strange return type. Simpler `Collection<Function>` would be more better, if found sufficient.